javascripts/controllers/contacts/ViewCtrl.js

```diff
@@ -21,4 +21,23 @@ app.controller("ViewCtrl", function($location, $rootScope, $scope, contactServic
 			console.log("error in deleteContact", err);
 		});
 	};
+
+
+
+	$scope.favoriteContact = (contactId) => {
+		contactId.uid = $rootScope.uid;
+		contactId.isFavorite = false;
```

Comment:
@Junochop - a couple things. 1) Your isFavorite is always being set to false, which I imagine is not what you want. 2) On line 31, you are calling createContactObject, which needs a whole contact, not just the ID